QuotesApp/Services/RevenueCat/PurchaseManager.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  PurchaseManager.swift
+//  QuotesApp
+//
+//  Created by Mateusz Grudzień on 18/05/2025.
+//
+
+import Foundation
+import RevenueCat
+import DependencyInjection
+
+protocol PurchaseManagerInterface {
+    func checkPremiumAction() async -> Bool
+}
+
+final class PurchaseManager: PurchaseManagerInterface {
+    @Injected private var coreDataManager: CoreDataManagerInterface
+    
+    enum Configuration {
+        static let quoteLimit = 3
+    }
+    
+    init() {
+        Purchases.logLevel = .debug
+        Purchases.configure(withAPIKey: Constants.paywallConfigurationKey)
+    }
+    
+    func checkPremiumAction() async -> Bool {
+        let hasPremium = await checkFullAccess()
+        let quotesCount = coreDataManager.fetchAllQuotes().count
+        
+        return hasPremium || quotesCount < Configuration.quoteLimit
```

Comment:
```suggestion
        return hasPremium || quotesCount <= Configuration.quoteLimit
```
Czy tutaj nie powinno być mniejsze lub równe?